src/components/test-taking/TestTakingClient.tsx

```
@@ -0,0 +1,784 @@
+
+  // Check if all required questions in current category are answered
+  const requiredQuestionsInCategory = currentCategory.items.filter(
+    (item) => item.isRequired !== false,
+  );
+  const requiredAnsweredCount = requiredQuestionsInCategory.filter(
+    (item) => responses[item.question.id] !== undefined,
+  ).length;
+  const canProceedToNext =
+    requiredAnsweredCount === requiredQuestionsInCategory.length;
+
+  React.useLayoutEffect(() => {
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)